pkg/dataobj/metastore/object.go

```diff
@@ -399,9 +401,63 @@ func (m *ObjectMetastore) listObjects(ctx context.Context, path string, start, e
 	if err != nil {
 		return nil, err
 	}
+
+	// Then we iterate over index objects based on the new format.
+	predicate := indexpointers.TimeRangeRowPredicate{
+		Start: start.UTC(),
+		End:   end.UTC(),
+	}
+	err = forEachIndexPointer(ctx, object, predicate, func(indexPointer indexpointers.IndexPointer) {
+		objectPaths = append(objectPaths, indexPointer.Path)
+	})
+	if err != nil {
+		return nil, err
+	}
+
 	return objectPaths, nil
 }
 
+func forEachIndexPointer(ctx context.Context, object *dataobj.Object, predicate indexpointers.RowPredicate, f func(indexpointers.IndexPointer)) error {
+	var reader indexpointers.RowReader
+	defer reader.Close()
+
+	buf := make([]indexpointers.IndexPointer, 1024)
+
+	for _, section := range object.Sections() {
+		if !indexpointers.CheckSection(section) {
```

Comment:
this should use the .Filter() approach. It was only added recently so I don't think I used it for the other forEach methods either - my bad!